Fix endermite spawn position

Suisuroru opened the pull request:
Origin issue: https://github.com/LeavesMC/Leaves/issues/631
Origin PR: https://github.com/LeavesMC/Leaves/pull/632
Origin license: [GPL3](https://github.com/LeavesMC/Leaves/blob/master/licenses/GPL.md)

record in [wiki](https://minecraft.wiki/w/Ender_Pearl): An ender pearl has a 5% chance to spawn an endermite when it lands. This is the only way through which endermites can spawn, without using cheats. The endermite spawns at the player's position when the pearl lands‌[Java Edition only], or at the pearl's landing site‌[Bedrock Edition only].

in the past the endermites will spawn at the position player teleported instead of old position becuse craftbukkit move teleport behind to create event, so it spawn in wrong place.

lynxplay commented on paper-server/patches/sources/net/minecraft/world/entity/projectile/ThrownEnderpearl.java.patch:
```diff
@@ -1,9 +1,12 @@
                  Vec3 vec3 = this.oldPosition();
                  if (owner instanceof ServerPlayer serverPlayer) {
                      if (serverPlayer.connection.isAcceptingMessages()) {
++                        // Paper start - fix endermite spawn position
```
Way too large of a diff marker, naming of the variables is generally also off, `preTeleportX` maybe?
Move it into the CraftBukkit comment and remove the paper comment.

lynxplay commented on paper-server/patches/sources/net/minecraft/world/entity/projectile/ThrownEnderpearl.java.patch:
```diff
@@ -14,8 +17,10 @@
+-                                endermite.snapTo(owner.getX(), owner.getY(), owner.getZ(), owner.getYRot(), owner.getXRot());
 -                                serverLevel.addFreshEntity(endermite);
++                                endermite.snapTo(x, y, z, yaw, pitch);
++                                // Paper end - fix endermite spawn position
```
`endermite.snapTo(preTeleportX, preTeleportY, preTeleportZ, preTeleportYaw, preTeleportPitch); // Paper - spawn endermite at pre teleport position as teleport has been moved up.`

lynxplay reviewed (changes requested):
Thank you for the port of the commit and welcome to paper 🥳 

The merger needs to ensure the author of the commit correctly credits the original author and move you as a co-author instead.

A bit of feedback on the diff markers.

Lumine1909 committed:
Fix endermite spawn position

The player teleport has been moved up in ThrownEnderpearl to prevent
endermite spawning in case of a cancelled teleport. This however placed
the player at the teleport destination prior to the endermite spawn
logic, creating the entity at the wrong location.

The commit stores the location of the player pre teleportation and uses
said location for spawning the endermite instead.

Co-authored-by: Helvetica Volubi <suisuroru@blue-millennium.fun>
Reviewed-in: https://github.com/PaperMC/Paper/pull/12912
